airo-typing/airo_typing/__init__.py

```diff
@@ -69,11 +69,67 @@
 shorthand notation is ^C T^B_A, where C is the frame the velocity is measured in, B is the frame the velocity is expressed in.
 """
 
-# manipulator types
+#####################
+# Manipulator types #
+#####################
 
 JointConfigurationType = np.ndarray
 """an (N,) numpy array that represents the joint angles for a robot"""
 
+#########################
+# Single-arm path types #
+#########################
+JointPathType = List[JointConfigurationType]
+""" a list of joint configurations that describe a path in joint space"""
+
+JointPathArrayType = np.ndarray
+""" a (N, DoFs) numpy array that represents a path in joint space"""
+
+PosePathType = List[HomogeneousMatrixType]
+""" a list of homogeneous matrices that describe a path in cartesian space"""
+
+PosePathArrayType = np.ndarray
+""" a (N, 4, 4) numpy array that represents a path in cartesian space"""
+
+TimePathType = List[float]
+""" a list of times that describe a path in time, must be monotonically increasing"""
+
+TimePathArrayType = np.ndarray
+""" a (N,) numpy array that represents a path in time, must be monotonically increasing"""
+
+#######################
+# Dual-arm path types #
+#######################
+DualJointConfigurationType = np.ndarray
+""" a (2 * DoFs,) numpy array, result of np.hstack((left_joints, right_joints))"""
+
+DualJointPathType = List[DualJointConfigurationType]
+""" a list of dual arm joint configurations that describe a path in joint space"""
+
+DualJointPathArrayType = np.ndarray
+""" a (N, 2 * DoFs) array, result of np.hstack((left_joint_path, right_joint_path))"""
+
+#######################
+# Single-arm types #
+#######################
+ForwardKinematicsType = Callable[[JointConfigurationType], HomogeneousMatrixType]
+""" a function that computes the forward kinematics of a given joint configuration"""
+
+InverseKinematicsType = Callable[[HomogeneousMatrixType], List[JointConfigurationType]]
```

Comment:
`InverseKinematicsFunctionType` maybe? To clearly indicate that it is a callable function